test/unit/specs/DefaultSlideTemplate.spec.js

```diff
@@ -2,272 +2,356 @@ import Vue from 'vue'
-      ],
-      title: {
-        content: 'Sherbrooke Event Title',
-        fontColor: 'Blue',
-        fontSize: 'Large',
-        fontStyle: 'Normal',
-        fontWeight: 'Oblique'
-      },
-      description: {
-        content: 'Sherbrooke event slide body. Lorem ipsum dolor sit amet, pri veniam forensibus id. Vis maluisset molestiae id ad semper lobortis cum. At impetus detraxit incorrupte usu, repudiare assueverit ex eum, ne nam essent vocent admodum.',
-        fontColor: 'Red',
-        fontSize: 'Small',
-        fontStyle: 'Normal',
-        fontWeight: 'Lighter'
-      },
-      date: {
-        content: '2018-03-07',
-        fontColor: 'Green',
-        fontSize: 'Medium',
-        fontStyle: 'Italic',
-        fontWeight: 'Bolder'
-      },
-      time: {
-        content: '11:05 AM',
-        fontColor: 'Yellow',
-        fontSize: 'Smaller',
-        fontStyle: 'Normal',
-        fontWeight: 'Bold'
-      },
-      meta: {
-        template: 'DefaultSlideTemplate',
-        timeout: '90',
-        repeatable: true,
-        startDate: '2018-07-25',
-        endDate: '2018-09-01'
-      }
-    }
-  }
+  let vm, Constructor // <- objects to be redefined
 
-  // reset the test case
-  beforeEach('Reset the contents to the preset style of props', function () {
-    // pass data into propsData
-    vm = new Constructor({ propsData: props })
+  before(function () {
+    Constructor = Vue.extend(DefaultSlideTemplate)
   })
 
-  it('Title contents should be rendered correctly', function () {
-    vm.$mount()
-    /* test contents in the title */
-    const title = vm.$el.querySelector('h2.display-3')
-    expect(vm.$props.slide.title.content).to.equal('Sherbrooke Event Title')
-    expect(title.textContent).to.equal(vm.$props.slide.title.content)
-    // TO DO
-    expect(vm.$props.slide.title.fontColor).to.equal('Blue')
-    // TO DO
-    expect(vm.$props.slide.title.fontSize).to.equal('Large')
-    // TO DO
-    expect(vm.$props.slide.title.fontStyle).to.equal('Normal')
-    // TO DO
-    expect(vm.$props.slide.title.fontWeight).to.equal('Oblique')
-    // TO DO
-  })
+  describe('Test slide content', function () {
+    let mockProps
 
-  it('Time content should be rendered corrctly', function () {
-    vm.$mount()
-    /* Test the contents in the header */
-
-    /* testing time */
-    const header = vm.$el.querySelector('h2.display-2')
-    expect(vm.$props.slide.time.content).to.equal('11:05 AM')
-    expect(header.textContent).to.contain(vm.$props.slide.time.content)
-    // TO DO
-    expect(vm.$props.slide.time.fontColor).to.equal('Yellow')
-    // TO DO
-    expect(vm.$props.slide.time.fontSize).to.equal('Smaller')
-    // TO DO
-    expect(vm.$props.slide.time.fontStyle).to.equal('Normal')
-    // TO DO
-    expect(vm.$props.slide.time.fontWeight).to.equal('Bold')
-    // TO DO
-  })
+    before(function () {
+      /* definition of props to be passed down */
+      mockProps = {
```

Comment:
can we hoist this mock props object to the spec scope of the describe functions call

Reply:
done